backend/src/api/mining/mining.ts

```diff
@@ -633,6 +645,17 @@ class Mining {
       default: return 86400 * scale;
     }
   }
+
+  // Finds the oldest block in a consecutive chain back from the tip
+  // assumes `blocks` is sorted in ascending height order
+  private getOldestConsecutiveBlock(blocks: DifficultyBlock[]): DifficultyBlock {
```

Comment:
I feel this this duplicate is not necessary. Isn't `BlocksRepository::$getOldestConsecutiveBlock()` serving the same purpose?

Reply:
yes, but that makes another expensive DB query to fetch all of the blocks again

Reply:
maybe we can pull this out into a utility function we can also call from `BlocksRepository::$getOldestConsecutiveBlock()`, to avoid the code duplication?

Reply:
> maybe we can pull this out into a utility function we can also call from `BlocksRepository::$getOldestConsecutiveBlock()`, to avoid the code duplication?

That seems a good way yes 👍🏻